src/components/login/login.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from "react";
+import { connect } from "react-redux";
+import { toast } from "react-toastify";
+import { IRootState } from "reducers";
+import { Button, Checkbox, Form } from "semantic-ui-react";
+import { login } from "../../actions/authentication.actions";
+import { goToUrl } from "../../store/routerHistory";
+import "./login.scss";
+
+interface DefaultState {
+    privateKey: string;
+    publicKey: string;
+}
+
+const mapStateToProps = (state: IRootState, props: any) => ({
+    currentUser: state.authentication.currentUser,
+    isLoading: state.authentication.isLoading,
+});
+
+const mapDispatchToProps = {
+    login,
+};
+
+type Props = ReturnType<typeof mapStateToProps> & typeof mapDispatchToProps;
+
+class Login extends React.Component<Props, DefaultState> {
+    constructor(props: any) {
+        super(props);
+
+        this.state = {
+            privateKey: "",
+            publicKey: "",
+        };
+    }
+
+    public handlePublicKeyChange = (element) => {
+        this.setState({
+            publicKey: element.target.value,
+        });
+    };
+
+    public handlePrivateKeyChange = (element) => {
+        this.setState({
+            privateKey: element.target.value,
+        });
+    };
+
+    public login = () => {
```

Comment:
Please conform to the React naming of `handle` in front of the actual name, so: `handleLogin`.
Please apply this throughout the entire application.